src/tree/JobTreeItem.ts

```diff
@@ -0,0 +1,103 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Octokit } from '@octokit/rest';
+import { ActionsGetJobForWorkflowRunResponseData } from '@octokit/types';
+import * as moment from 'moment';
+import * as prettyMs from 'pretty-ms';
+import { AzExtTreeItem, AzureParentTreeItem, IActionContext, TreeItemIconPath } from "vscode-azureextensionui";
+import { createOctokitClient } from '../commands/github/createOctokitClient';
+import { Conclusion, Status } from '../gitHubTypings';
+import { convertConclusionToVerb, convertStatusToVerb, getRepoFullname } from '../utils/gitHubUtils';
+import { localize } from '../utils/localize';
+import { treeUtils } from "../utils/treeUtils";
+import { ActionTreeItem } from './ActionTreeItem';
+import { IAzureResourceTreeItem } from './IAzureResourceTreeItem';
+import { StepTreeItem } from './StepTreeItem';
+
+export class JobTreeItem extends AzureParentTreeItem implements IAzureResourceTreeItem {
+
+    public static contextValue: string = 'azureStaticJob';
+    public readonly contextValue: string = JobTreeItem.contextValue;
+    public parent: ActionTreeItem;
+    public data: ActionsGetJobForWorkflowRunResponseData;
+
+    constructor(parent: ActionTreeItem, data: ActionsGetJobForWorkflowRunResponseData) {
+        super(parent);
+        this.data = data;
+    }
+
+    public get iconPath(): TreeItemIconPath {
+        return treeUtils.getActionIconPath(this.status, this.conclusion);
+    }
+
+    public get id(): string {
+        return this.data.id.toString();
+    }
+
+    public get name(): string {
+        return this.data.name;
+    }
+
+    public get label(): string {
+        return this.name;
+    }
+
+    public get description(): string {
+        if (this.data.conclusion !== null) {
+            const elapsedMs: number = this.completedDate.getTime() - this.startedDate.getTime();
+            return `${convertConclusionToVerb(this.conclusion)} ${moment(this.completedDate).fromNow()} in ${prettyMs(elapsedMs)}`;
```

Comment:
> Actually, moment is writing complete phrases like 21 days ago or 21 minutes ago, so I'm not sure how we would localize that unless moment has language support.

Moment supports localization. However, I don't think "pretty-ms" does. Even if it did - my original comment was that you're putting too much stuff in the description and I stand by that. The biggest reason I love VS Code is because of how lightweight/simple it is - and I want our extensions to reflect that. Pick only the most important information to display and throw the rest in a "View Properties" command

Reply:
I say we at least keep how long ago it ran (i.e. 4 days ago).  I don't know if we need the verbs (like completed for instance) in there since that's kind of the point of icons (though I'm pretty sure that's not accessible, so I dunno).  

I personally like the descriptions, and to your point about being lightweight/simple, while I agree, this level of description only appears if they drill down pretty far into the Actions node, in which case I feel like the user would be trying to get more detailed information about the steps.

The data object (the GitHub response type) doesn't include the times in a human friendly format (they just have completed/started timestamps) so I see some value in displaying that in a easily read manner.

Reply:
I'm fine with how long ago it ran and the conclusion/status (because the icons _aren't_ accessible, as you mentioned). The conclusion/status is most important (gotta know if it worked) and how long ago is second most important (if it ran a month ago, I might need to update my site). I just don't see as much value in the elapsed time - why do I care if it took a minute or an hour?

> this level of description only appears if they drill down pretty far into the Actions node, in which case I feel like the user would be trying to get more detailed information about the steps.

so if it looks gross already - we're allowed to make it even more gross? It's not that far down :P

> The data object (the GitHub response type) doesn't include the times in a human friendly format (they just have completed/started timestamps) so I see some value in displaying that in a easily read manner.

Doesn't mean you have to display it in the description. There's no reason the "properties" object we display has to be the exact object we get from GitHub. We can modify and make it look better as much as we want (still needs to be localizable, though - and out of scope for this PR).